battlepass/src/types.rs

```diff
@@ -0,0 +1,29 @@
+use frame_support::pallet_prelude::*;
+use codec::MaxEncodedLen;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
+// #[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+pub enum BattlepassState {
+	Inactive = 0,
+	Active = 1,
+}
+impl Default for BattlepassState {
+	fn default() -> Self {
+		Self::Active
+	}
+}
+
+
+#[derive(Encode, Decode, Default, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+// #[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+pub struct Battlepass<Hash, AccountId, BlockNumber, BoundedString> {
+	pub creator: AccountId,
+	pub org_id: Hash,
+	pub name: BoundedString,
+	pub cid: BoundedString,
+    pub state: BattlepassState,
```

Comment:
Usually, we have separate storage for the state (ex. `MemberStates`, `OrgStates`). I'm not sure what `ActiveBattlepassByOrg` is, but storing the state in the struct and storing it in the separate storage is suboptimal since you need to update both.

Reply:
Will create storage for states, like `battlepass_id -> BattlepassState`
ActiveBattlepassByOrg - is to check which is a currently active battlepass for the Org (`org_id -> active battlepass_id`). So the Active state will be stored in two places, but it will bring more benefits in terms of storage and usability, I suppose.

Reply:
you can query the active battlepass either via the graph as a client or by filtering for active pass as a first class citizen (the pallet), what is the usecase for the pallet? it would only operate per id on a specific battlepass and reject if the operation is not allowed, e.g. not active, not owner

e.g. `org` -> `vec<battlepass>`

Reply:
Example of use case: upon creation of new Battlepass for the Org, we need to check if there is an Active Battlepass (current season) already. Org may have multiple Battlepasses, but only one can be Active. So having a mapping `org` -> `active battlepass` is more convenient and faster than working with vec.